modules/core/react/lib/InputProvider.tsx

```diff
@@ -345,12 +370,16 @@ export default class InputProvider extends React.Component<InputProviderProps, I
   }
 
   render() {
-    if (this.state.nested) {
+    const {container} = this.props;
+    if (this.state.nested || container) {
       return this.props.children;
     }
 
     const intent = this.provideIntent ? this.state.currentIntent : null;
-    return (
+
+    return container ? (
+      container
+    ) : (
       <div
         className={containerClass}
         ref={this.ref}
```

Comment:
Now that we always have a reference to the container, we no longer need this div or ref.